Geometry_Engine/Query/IsContaining.cs

```diff
@@ -709,6 +709,158 @@ public static bool IsContaining(this Cuboid cuboid, IGeometry geometry, bool acc
+        [Input("closedVolume", "A collection of Polylines outlining each of the faces of the closed volume to check with.")]
+        [Input("planes", "Planes corresponding to each panel for intersection calculations.")]
+        [Input("boundingBox", "The bounding box of the collection of panels.")]
+        [Input("point", "The point to check to see if it is contained within the bounds of the panels.")]
+        [Input("acceptOnEdges", "Decide whether to allow the point to sit on the edge of the panel, default false.")]
+        [Input("tolerance", "Distance tolerance to use to determine intersections.")]
+        [Output("isContaining", "True if the point is contained within the bounds of the panels, false if it is not for each point provided.")]
+        private static bool IsContaining(this List<Polyline> closedVolume, List<Plane> planes, BoundingBox boundingBox, Point point, bool acceptOnEdges = false, double tolerance = BH.oM.Geometry.Tolerance.Distance)
+        {
+            //Return if point is null even without checking boundingBox.IsContaining(point)
+            if (point == null)
+            {
+                BH.Engine.Base.Compute.RecordError("Cannot query if a collection of panels contains a point if the point is null.");
+                return false;
+            }
+
+            if (planes.Any(x => x == null))
+            {
+                BH.Engine.Base.Compute.RecordError("Cannot query if a provided plane is null.");
+                return false;
+            }
+
+            if (!BH.Engine.Geometry.Query.IsContaining(boundingBox, point, true, tolerance))
+                return false;
+
+            Vector rVec;
+            bool isInPlane = false;
+            do
+            {
+                int seed = 1;
+                 rVec = BH.Engine.Geometry.Create.RandomVector(seed); 
+                
+                isInPlane = false;
+                foreach (Plane p in planes)
+                    isInPlane = isInPlane || rVec.IsInPlane(p, tolerance);
+
+                if (isInPlane)
+                {
+                    seed += 1;
+                }
+            }
+            while (isInPlane);
+
+            //Use a length longer than the longest side in the bounding box. Change to infinite line?
+            Line line = new Line() { Start = point, End = point.Translate(rVec * (((boundingBox.Max - boundingBox.Min).Length()) * 10)) };
+
+            //Check intersections
+            List<Point> intersectPoints = new List<Point>();
+
+            for (int x = 0; x < planes.Count; x++)
+            {
+                Point intPt = PlaneIntersection(line, planes[x], true, tolerance);
+                if (intPt == null)
+                    continue;
+                Polyline pLine = closedVolume[x];
+
+                if  (BH.Engine.Geometry.Query.IsContaining(boundingBox, intPt, true, tolerance) && BH.Engine.Geometry.Query.IsContaining(pLine, new List<Point> { intPt }, false, tolerance))
```

Comment:
I am thinking abstract about the problem in the context of call to `IsContaining(Polyline)` query, we have 3 interesting intersection scenarios:
- point inside solid, ray not hitting any edge
- point inside solid, ray hitting edges
- point outside solid, ray hitting edges

The 1st will always yield correct results, 2nd will yield wrong results if `acceptOnEdges` is set to `false` and the ray leaves the solid through the edges, while 3rd will return false if `acceptOnEdges` is set to `true` and the ray hits an edge.

The above makes perfect sense if you look at how the method for `Polyline` was written:
https://github.com/BHoM/BHoM_Engine/blob/851d1acd212c1aa4123d24b4f3749154af958146/Geometry_Engine/Query/IsContaining.cs#L337

...writing similar for 2d panels instead of lines sounds like lots of fun, so I started thinking how to avoid it - what came to my mind is that you could add one more check against the random vector to make sure that the ray does not hit the outlines of any panel, and if it does, then pick another random. What do you think?

Follow up thought: would be good to start from checking if the point itself does not lie on the edge - can help you skip a few heave computations + will let you avoid infinite loop in case of implementation of the suggestion above.

Reply:
Suggest this is created as a separate follow up issue,

Reply:
Resolved per latest commit from @pawelbaran

Reply:
Resolved with latest changes.